src/ngraph/event_tracing.cpp

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "event_tracing.hpp"
+#include "nlohmann/json.hpp"
+
+using namespace std;
+
+mutex ngraph::Event::s_file_mutex;
+ofstream ngraph::Event::s_event_log;
+
+namespace ngraph
+{
+    void ngraph::Event::write_trace(const ngraph::Event& event)
+    {
+        lock_guard<mutex> lock(s_file_mutex);
+        if (!is_tracing_enabled())
+            return;
+
+        static bool initialized = false;
+        if (!initialized)
+        {
+            // Open the file
+            s_event_log.open("ngraph_event_trace.json", ios_base::trunc);
+            s_event_log << "[\n";
+            s_event_log << event.to_json() << "\n";
+            initialized = true;
+            return;
+        }
+
+        s_event_log << ",\n";
+        s_event_log << event.to_json() << "\n" << flush;
```

Comment:
Remove the `flush`

Reply:
The flush is there by design. If the software using these Events crashe (which often does) then there is a trace of the events preserved in the file. So we need the flush here.

Reply:
What is the performance impact of the flush? That is my concern

Reply:
Based on the measurements - it's in the micro seconds range. Considering where these Events are going to be used - the cost is insignificant. If we do see any performance issues then to would be easy to convert these to a circular queue/atomic instructions to minimize any overhead. But in a future PR if needed.